### ISSUE TITLE
Support for Mapping, MutableMapping, Sequence, etc?

### ISSUE THREAD

**Issue by dargueta** [python-desert/desert#140]
Marshmallow supports serializing arbitrary sequences and mappings, so long as they provide the correct interface. Unfortunately, desert doesn't seem to allow generic types out of the box:

The following crashes:

```py
import dataclasses

@dataclasses.dataclass
class D:
    foo: Sequence[str]

schema = desert.schema(D)
```

Traceback:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/site-packages/desert/__init__.py", line 24, in schema
    return desert._make.class_schema(cls, meta=meta)(many=many)
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/site-packages/desert/_make.py", line 130, in class_schema
    attributes[field.name] = field_for_schema(
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/site-packages/desert/_make.py", line 291, in field_for_schema
    nested = forward_reference or class_schema(typ)
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/site-packages/desert/_make.py", line 105, in class_schema
    raise desert.exceptions.UnknownType(
desert.exceptions.UnknownType: Desert failed to infer the field type for typing.Sequence[str].
Explicitly pass a Marshmallow field type.
```

Can we add support for these? It seems like the changes would need to be made [here](https://github.com/python-desert/desert/blob/main/src/desert/_make.py#L241-L257) and not too much work.

### Environment

Python 3.8.6

* desert: 2020.11.18
* marshmallow: 3.13.0
* marshmallow-enum: 1.5.1
* marshmallow-union: 0.1.15.post1

**Comment by altendky** [comment]
I might be forgetting something, but it strikes me as reasonable.  I'm not looking for more work to put on my plate right now, so I may not get to this soon.  Any interest in putting in a PR with tests to try and uncover any corners we should think about?

**Comment by dargueta** [comment]
I don't have time right now either, unfortunately. I can't promise I'll get around to it within this decade but I'll try. 🙂

**Comment by dargueta** [comment]
@altendky I've finally implemented this in #164.
